pkg/controller/registry/reconciler/grpc.go

```diff
@@ -287,17 +282,8 @@ func updatePodByDigest(updatePod *corev1.Pod, servingPods []*corev1.Pod) bool {
 	return false
 }
 
-func getPodImageID(pod *corev1.Pod) string {
-	var podImageID string
-
-	if pod.Status.ContainerStatuses != nil && len(pod.Status.ContainerStatuses) > 0 {
-		podImageID = pod.Status.ContainerStatuses[0].ImageID
-	} else {
-		logrus.WithField("CatalogSource", pod.GetName()).Warn("pod status unknown")
-	}
-
-	logrus.WithField("CatalogSource", pod.GetName()).Infof("ImageID %s", podImageID)
-	return podImageID
+func imageID(pod *corev1.Pod) string {
+	return pod.Status.ContainerStatuses[0].ImageID
```

Comment:
I think this was one of issues we did bump into the last time that when the ImageID is not there, this will panic out due to null value.

Reply:
Right so that's why I call https://github.com/operator-framework/operator-lifecycle-manager/pull/1723/files#diff-8f8ed949ede33539a8c472480f3d4f25R330 first to ensure its non-nil. This makes this function extremely simple, almost unnecessary really